File: saec-frontend/src/services/profileService.js
import api from './api';

const profileService = {
  /**
   * GET /api/profile/me
   * Devuelve el perfil completo del usuario autenticado.
   * @returns {Promise<PerfilUsuarioDto>}
   */
  async getMyProfile() {
    const { data } = await api.get('/profile/me');
    return data;
  },

  /**
   * PATCH /api/profile/me
   * Actualiza teléfono y correo electrónico del usuario autenticado.
   * @param {{ email: string, telefono: string|null }} payload
   * @returns {Promise<PerfilUsuarioDto>}
   */
  async updateMyContact(payload) {
    const { data } = await api.patch('/profile/me', payload);
    return data;
  },
};

export default profileService;

/**
 * @typedef {Object} CentroPerfilDto
 * @property {number}  idCentro
 * @property {string}  nombre
 * @property {string}  ciudad
 * @property {string|null} direccion
 * @property {string|null} telefono
 */

/**
 * @typedef {Object} PerfilUsuarioDto
 * @property {number}  idUsuario
 * @property {string}  nombre
 * @property {string}  apellido
 * @property {string}  nombreCompleto
 * @property {string}  email
 * @property {string|null} documentoIdentidad
 * @property {string}  rol
 * @property {string|null} descripcionRol
 * @property {CentroPerfilDto[]} centros
 * @property {boolean} activo
 * @property {string}  createdAt    ISO-8601
 * @property {string|null} ultimoAcceso ISO-8601
 * @property {string|null} fotoPerfil
 * @property {string|null} especialidadCargo
 * @property {string|null} telefono
 * @property {string|null} ciudad
 * @property {string|null} direccion
 */

File: saec-backend/src/main/java/com/saec/service/ProfileService.java
package com.saec.service;

import com.saec.dto.profile.ActualizarContactoDto;
import com.saec.dto.profile.PerfilUsuarioDto;
import org.springframework.security.core.userdetails.UserDetails;

public interface ProfileService {

    /**
     * Devuelve el perfil completo del usuario actualmente autenticado.
     *
     * @param userDetails principal Spring Security del usuario autenticado
     * @return DTO con toda la información del perfil
     */
    PerfilUsuarioDto getMyProfile(UserDetails userDetails);

    /**
     * Actualiza teléfono y/o correo electrónico del usuario autenticado.
     *
     * @param userDetails principal Spring Security del usuario autenticado
     * @param dto         campos a actualizar
     * @return perfil actualizado
     */
    PerfilUsuarioDto updateMyContact(UserDetails userDetails, ActualizarContactoDto dto);
}
